xpk.py

```diff
@@ -2660,17 +2646,20 @@ def run_gke_cluster_create_command(args, gke_control_plane_version: str) -> int:
   return 0
 
 
-def set_up_cluster_network_for_a3(args) -> int:
-  """Set up GKE Cluster networks, subnets and firewall rules for A3.
-  Note: there are 4 NICs for GPU-GPU bw and 1 NIC for host in a A3 node
+def set_up_cluster_network_for_gpu(args) -> int:
+  """Set up GKE Cluster networks, subnets and firewall rules for A3/A3+.
+  Note: there are 4 NICs for GPU-GPU bw and 1 NIC for host in an A3 node,
+  and there are 8 NICs for GPU-GPU bw and 1 NIC for host in an A3+ node.
 
   Args:
     args: user provided arguments for running the command.
 
   Returns:
     0 if successful and 1 otherwise.
   """
-  for i in range(1, 5):
+  device_type = args.tpu_type if args.tpu_type else args.device_type
```

Comment:
Could we pass the SystemCharacteristics into`set_up_cluster_network_for_gpu` instead of checking the tpu/device_type args? Hoping to avoid duplication of this arg check in the code and propagate SystemCharacteristics instead